### PR TITLE
Updated overlays to use a settings parameter

### LINKED ISSUE #1175: Post-ruff minor issues
Now that all the core IPv8 logic has had its `ruff` issues fixed, including its types, I'm left with a list of minor issues that should be solved at some point:

 - [x] Both `TestBase` and `DiscoveryStrategy` should be turned into `Generic[SomeOverlayType]`. *WHY: Having to assure mypy that they were loaded with a particular type of overlay, at runtime, is a pain.* **Fixed in** #1181 and #1192
 - [ ] Create a `CommunityConfig` (or a dictionary) to store `Community` configurations. *WHY: We now have 6 arguments to the `Community` constructor and, not only is this triggering `PLR0913` everywhere, this leads to mass repetition of the default arguments.*
 - [ ] Related to the previous point: the docs should be updated accordingly.
 - [x] `PD011` is throwing false-positive `numpy` errors and should be disabled. *WHY: We don't use `numpy`.* **Fixed in** #1190
 - [x] The `RequestCache` should type its return values. *WHY: Currently we need to cast all of the results of `pop()` from the `RequestCache`.* **Fixed in** #1191
 - [x] We could shorten some types to be more descriptive. *WHY: We now have very long types which are (probably) correct but - for example - one `bytes` may describe a 20 byte infohash while the other is a 64 byte public key.* **Fixed in** #1197

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #1175

This PR:

 - Updates all `Overlay` instances to take a `settings` object instead of passing individual settings to the `__init__()`.

### CODE REVIEW

**Comment by qstokkink** [comment]
validate

**Comment by qstokkink** [comment]
validate

**Comment by qstokkink** [comment]
validate

File: doc/basics/testbase_tutorial_1.py [hunk 1]
```diff
@@ -26,16 +26,18 @@ def on_timeout(self) -> None:
         self.overlay.timed_out = True
 
 
+class MyCommunitySettings(CommunitySettings):
```
Comment (egbertbouman) [inline doc/basics/testbase_tutorial_1.py:29]: Shouldn't this be added to `MyCommunity.settings_class`?

**Comment by egbertbouman** [comment]
validate

File: doc/basics/testbase_tutorial_1.py (hunk 1 above)
Comment (qstokkink) [reply doc/basics/testbase_tutorial_1.py:28]: Hold on, this entire test class is not passing 😨

**Comment by qstokkink** [comment]
validate

File: doc/basics/testbase_tutorial_1.py (hunk 1 above)
Comment (qstokkink) [reply doc/basics/testbase_tutorial_1.py:28]: Strange. I wonder why the validation tests did not catch that.. 🤔

**Review by egbertbouman — APPROVED** [review]

**Pull request merged**